text_mining.py

```diff
@@ -0,0 +1,357 @@
+    # Part 1 -> Pickles Twitter 'user instances' of all epl_players
+    #           listed in a Twitter List(EPL Players, Official BBC list)
+    players_data = 'epl_players.pickle'
+    if not exists(players_data):
+        save_epl_members_list(players_data, api)
+    # list of all player instances
+    all_players = load_epl_members_list(players_data)
+
+    # --------------------------------------------------------------------------
+    # Part 2 -> Pickles Twitter 'status instances' for all 'user instances'
+    for player in all_players:
+        #  pickles status data for users with unprotected timelines
+        status_data = "players_status/" + player.name
+        if not exists(status_data) and not player.protected:
+            save_tweets(player.id, player.name, api)
+
+    # --------------------------------------------------------------------------
+    # Part 3 -> Pickles tweet texts of status instances(unprotected timeline)
+    # Texts are processed so that tags, hashtags, htmls are elimimnated.
+    # Only Tweets from the first day of EPL16-17 season is saved.
+    # The maximum number of status instances for each user insnaces is 200
+
+    # First Day of EPL16-17
+    START_TIME = datetime.datetime.strptime("Sat Aug 13 00:00:00 +0000 2016",
+                                            '%a %b %d %X %z %Y')
+    tweets_data = 'epl_players_tweets.pickle'
+    if not exists(tweets_data):
+        save_tweets_texts(tweets_data, all_players)
+    tweet_texts = load_tweets_texts(tweets_data)  # Processed tweet texts
+
+    # --------------------------------------------------------------------------
+    # Part 4 -> Conducts Sentimental Analysis(NLTK)
+    #           with tweet texts of all user instnaces with unprotected
+    #           timelines. (Tweets of EPL Players with open timelines)
+    nltk_results = all_players_nltk_analysis(all_players, tweet_texts)
+
+    # --------------------------------------------------------------------------
+    # Part 5 -> Regroups the NLTK results by EPL Clubs.
+    #           Pickles current club of of a player from Wikipedia
+    #           by using Wikipedia and BeautifulSoup package.
+    #           (as a dictoinary that maps a player to a club)
+    clubs_data = 'epl_clubs_players.pickle'
+    if not exists(clubs_data):
+        save_clubs_dictionary(clubs_data, all_players)
+    clubs_dict = load_clubs_dictionary(clubs_data)
+    club_analysis_results = club_nltk_analysis(nltk_results, clubs_dict)
+    # --------------------------------------------------------------------------
+    # Part 6 -> Displays results sorted by different sentiments of NLTK
+    #           Keys : compound, neg, neu, pos
+    print(sorted_by_key(club_analysis_results, 0))  # key = compound
+    print(sorted_by_key(club_analysis_results, 1))  # key = neg
+    print(sorted_by_key(club_analysis_results, 2))  # key = neu
+    print(sorted_by_key(club_analysis_results, 3))  # key = pos
```

Comment:
Really nice step-by-step guide. But I think you would have been better off putting all this script as a separate python file.